feat: Update pull request template (#52)
nginx/documentation: .github/pull_request_template.md

### Checklist

Before merging a pull request, run through this checklist and mark each as complete.

- [] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [] I have signed the [F5 Contributor License Agreement (CLA)](https://github.com/f5/.github/blob/main/CLA/cla-markdown.md)
- [] I have ensured that documentation content adheres to [the style guide](/templates/style-guide.md)
- [] If the change involves potentially sensitive changes, I have assessed the possible impact
- [] If applicable, I have added tests that prove my fix is effective or that my feature works
- [] If applicable, I have checked that any relevant tests pass after adding my changes
- [] I have updated any relevant documentation ([`README.md`](/README.md) and [`CHANGELOG.md`](/CHANGELOG.md))
- [] I have rebased my branch onto main
- [] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

> Potentially sensitive changes include anything involving code, personally identify information (PII), live URLs or significant amounts of new or revised documentation.

Reviewer: maybe:

```suggestion
- [] Pay special attention to the following types of changes: code, personally identify information (PII), URLs (watch for typosquatting), or significant amounts of new or revised documentation.
```

Author: The sentence is intended to support the existing item on line 27: this makes it somewhat redundant.

I'm banking on the idea that we give better explanations of these issues in the style guide, as reducing the amount of reading fatigue on a template some people might be exposed to daily feels paramount.

Reviewer: > The sentence is intended to support the existing item on line 27: this makes it somewhat redundant.

- I'd be good if it were a sub-item of line 27. When our security engineers looked at this repo, they wanted us to pay special attention to these items.

Reviewer: While I'd love to see automation, I think this is beyond anything we could put in our style guide. (If you can show me how, I'd appreciate it!.)

Author: I'm not really sure why automation is part of this discussion.

> - [] If the change involves potentially sensitive changes, I have assessed the possible impact
> - [] If applicable, I have added tests that prove my fix is effective or that my feature works
> - [] If applicable, I have checked that any relevant tests pass after adding my changes
> - [] I have updated any relevant documentation ([`README.md`](/README.md) and [`CHANGELOG.md`](/CHANGELOG.md))
> - [] I have rebased my branch onto main
> - [] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork
> 
> Potentially sensitive changes include anything involving code, personally identify information (PII), live URLs or significant amounts of new or revised documentation.

Reviewer: ```suggestion
- [] If applicable, I have added tests that prove my fix is effective or that my feature works
- [] If applicable, I have checked that any relevant tests pass after adding my changes
- [] I have updated any relevant documentation ([`README.md`](/README.md) and [`CHANGELOG.md`](/CHANGELOG.md))
- [] I have rebased my branch onto main
- [] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

- [] If the change involves potentially sensitive changes, I have assessed the possible impact
   - These potentially sensitive changes include: 
      - Changes to code
      - Personally identifying information
      - URLs which may be affected by [typosquatting](https://en.wikipedia.org/wiki/Typosquatting)
```

Author: The driving effort of this PR was to fix the template, which was caused by the existed bulleted items breaking the formatting: this re-introduces the issue.

Reviewer 2: brainstorming: the more items the list has, the less likely I am to even read it.

There is a lot of scope creep in a PR that should have been a quick fix.

Reviewer: > There is a lot of scope creep in a PR that should have been a quick fix.

I disagree, but have moved it to a new issue, #89
